src/fileHandler.ts

```diff
@@ -0,0 +1,76 @@
+import * as vscode from 'vscode';
+import * as path from 'path';
+import * as fs from 'fs';
+import * as namespace from './namespace';
+
+function isCljFile(uri: vscode.Uri): boolean {
+  return uri.fsPath && ((uri.fsPath.endsWith(".clj")
+    || uri.fsPath.endsWith(".cljs")
+    || uri.fsPath.endsWith(".cljc")));
+}
+
+function isPrefix(parentPath: string, filePath: string): boolean {
+  const relative = path.relative(parentPath, filePath);
+  return !relative || (!relative.startsWith('..') && !path.isAbsolute(relative));
+}
+
+function resolveNsName(sourcePaths: string[], filePath: string): string {
+  if (!sourcePaths || !sourcePaths.length) {
+      return "";
+  }
+  for (const sourcePath of sourcePaths) {
+      if (isPrefix(sourcePath, filePath)) {
+          const relative = path.relative(sourcePath, path.dirname(filePath));
+          return relative.replace(/[\/\\]/g, ".")
+                         .replace("_", "-"); // Is such style acceptable?
+      }
+  }
+  return "";
+}
+
+function resolveFileName(filePath: string): string {
+  const fileName: string = path.basename(filePath);
+  const extName: string = path.extname(fileName);
+  return fileName.substring(0, fileName.length - extName.length)
+                 .replace("_", "-");
+}
+
+export default async function handleNewCljFiles(e: vscode.FileCreateEvent) {
+  const emptyFiles: vscode.Uri[] = [];
+  const textDocuments: vscode.TextDocument[] = [];
+  for (let file of e.files) {
+      if (!isCljFile(file)) {
+          continue;
+      }
+      const textDocument = await vscode.workspace.openTextDocument(file);
+      if (textDocument.getText()) { // Ignore the non-empty files.
+          continue;
+      }
+      emptyFiles.push(file);
+      textDocuments.push(textDocument);
+  }
+  if (!emptyFiles.length) {
+      return;
+  }
+  let sourcePaths: string[] = [];
+  const client = namespace.getSession();
+  if (client) {
+      const resp = await client.classpath();
+      if (resp && resp.classpath) {
+        sourcePaths = resp.classpath.filter(p => fs.existsSync(p) && fs.lstatSync(p).isDirectory());
+      }
+  }
+  for (let i = 0; i < emptyFiles.length; i++) {
+      const filename: string = resolveFileName(textDocuments[i].fileName);
+      const nsname: string = resolveNsName(sourcePaths, emptyFiles[i].fsPath);
+      const snippets: string[] = [];
+      if (nsname) {
+          snippets.push(`(ns ${nsname}.${filename}\$1)`);
+      } else {
+          snippets.push(`(ns ${filename}\$1)`);
```

Comment:
What is the point of the `\$1`?

Reply:
The point is to move the cursor to this position right after an ns form is created. It's needed because the next highly possible step is to add a require block. Since the cursor at the right position, you just need to hit the enter.

$1, $2 and so on are called [tabstops](https://code.visualstudio.com/docs/editor/userdefinedsnippets#_tabstops) and used in snippets.